sycl/doc/KernelParameterPassing.md

```diff
@@ -188,76 +202,56 @@ the non-wrapped array which is a member of the local capture object.
 ```C++
 static constexpr
 const kernel_param_desc_t kernel_signatures[] = {
-  //--- _ZTSZZ4mainENKUlRN2cl4sycl7handlerEE16->18clES2_E6Worker
+  //--- _ZTSZZ1fRN2cl4sycl5queueEENK3$_0clERNS0_7handlerEE6Worker
   { kernel_param_kind_t::kind_accessor, 4062, 0 },
-  { kernel_param_kind_t::kind_std_layout, 400, 32 },
+  { kernel_param_kind_t::kind_std_layout, 4, 32 },
+  { kernel_param_kind_t::kind_std_layout, 4, 36 },
+
 };
+
 ```
 
 <h4>The changes to device code made to support this extension, in pseudo-code:</h4>
 
 ```C++
 struct Capture {
-    Accessor outAcc;
-    int array[100];
+    sycl::accessor outAcc;
+    int array[num_items];
     () {
         // Body
     }
 }
 
-struct wrapper {
-    int array[100];
-};
 spir_kernel void caller(
-    __global int* AccData, // arg1 of Accessor init function
-    range<1> AccR1,        // arg2 of Accessor init function
-    range<1> AccR2,        // arg3 of Accessor init function
-    id<1> I,               // arg4 of Accessor init function
-    struct wrapper w_s     // Pass the array wrapped in a struct
+    __global int* AccData, // arg1 of accessor init function
+    range<1> AccR1,        // arg2 of accessor init function
+    range<1> AccR2,        // arg3 of accessor init function
+    id<1> I,               // arg4 of accessor init function
+    int p_array_0;         // Pass array element 0
+    int p_array_1;         // Pass array element 1
 )
 {
     // Local capture object
     struct Capture local;
 
     // Reassemble capture object from parts
     // Initialize array using existing clang Initialization mechanisms
-    local.array = w_s; 
+    local.array[0] = p_array_0;
+    local.array[1] = p_array_1; 
     // Call accessor�s init function
-    Accessor::init(&local.outAcc, AccData, AccR1, AccR2, I);
+    sycl::accessor::init(&local.outAcc, AccData, AccR1, AccR2, I);
 
     callee(&local, id<1> wi);
 }
 ```
 
-The sharp-eyed reviewer of `SemaSYCL.cpp` will notice that the array
-is actually double-wrapped in structs. This was done simply to preserve
-the interface to an existing function (`CreateAndAddPrmDsc`) which
-processes each kernel caller parameter as a capture object member. 
-The object being added to a list in `CreateAndAddPrmDsc` is `Fld`,
-which is expected to be a field of some struct. So a wrapped struct
-cannot be passed to this function. A double-wrapped struct is needed
-as shown below. This does not affect the generated code.
-
-```C++
-struct {
-  struct {
-    int array[100];
-  }
-}
-```
```

Comment:
Is this extra wrapping gone from the implementation? If not, I think this is important design detail and should be mentioned in the doc.